### PR TITLE
Tammy fisher

### PR DESCRIPTION
**Pull request by TammyLFisher**

### CODE REVIEW

File: assignments/callbacks.js [hunk 1]
```diff
@@ -25,31 +25,90 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
...
+ 
+
 function getLength(arr, cb) {
-  // getLength passes the length of the array into the callback.
+  return cb();
 }
+console.log(getLength(items, function () {
+  return items.length;
+}) );
+
```
Comment (dswhitely1) [inline assignments/callbacks.js]: Good evening.  I just wanted to show you the solution for this problem.  Can you go and refactor the rest of your answers to match this one.

```sh
function getLength(arr, cb) {
    return cb(arr.length);
}
getLength(items, function(length) {
   console.log(length);
}
```
If you have any further questions, feel free to reach out to your Team Lead or myself during the vacation week.  

Thanks,

Donald Whitely, TL, WEBPT8

File: assignments/array-methods.js [hunk 2]
```diff
@@ -56,28 +56,58 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
...
+    fullName.push(currentName);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+let allCaps = runners.map(function(nowallCaps) { 
+    return nowallCaps.first_name.toUpperCase()});
```
Comment (liam-mei) [inline assignments/array-methods.js]: With JS try to stick to using camel case.  so nowAllCaps would be preferred

File: assignments/callbacks.js [hunk 3]
```diff
@@ -25,31 +25,90 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
...
 
 /* STRETCH PROBLEM */
 
-function removeDuplicates(array, cb) {
+function removeDuplicates(items, alert) {
+
+  let unique = {};
+    items.forEach(function(i) {
+      items=>alert(items)
```
Comment (liam-mei) [inline assignments/callbacks.js]: what does this line do? 
`items=>alert(items)`

**Review by liam-mei — CHANGES_REQUESTED** [review]

**Comment by liam-mei** [comment]
it seems that you're getting the closure and array methods.  Choosing better parameter names might make it a bit more helpful though.

For the callbacks though, you're going to need to work on that.  Donald provided an excellent example and I hope you can refactor the rest of the problems to match it.

For the callbacks/higher order function.  What should be happening is that the higher order function/parent performs some kind of action to a parameter/array in this instance.  Then, returned value is used to invoke the callback function.

What you're doing is that in the higher order function, you're just invoking the callback and not passing it a parameter.  Then you're directly passing the argument into the cb function.

File: assignments/callbacks.js [hunk 4]
```diff
@@ -24,91 +24,58 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum','yo-yo', 'Gum'];
...
-  // getLength passes the length of the array into the callback.
+getLength(items, function(length) {
+  console.log(length);
+});
 
+// getLength passes the length of the array into the callback.
 
 function last(arr, cb) {
-  return cb();
+  return cb(arr.last);
```
Comment (liam-mei) [inline assignments/callbacks.js:37]: Here you're using the dot notation on an array.  On arrays, dot notation is to access methods/properties but arrays don't have a `last` property.

items.slice(-1)[0] will return you the last item of the array, but you're doing that on the function invocation and not inside the function declaration.

File: assignments/callbacks.js [hunk 5]
```diff
@@ -24,91 +24,58 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum','yo-yo', 'Gum'];
...
-
-  // multiplyNums multiplies two numbers and passes the result to the callback.
+last(items, function(last) {
+  last = items.slice(-1)[0];
+  console.log(last);
+});
+
+// last passes the last item of the array into the callback.
+function sumNums(x, y, cb) {
+  return x + y;
```
Comment (liam-mei) [inline assignments/callbacks.js:46]: Here, you're returning `x + y` but it is not passed into a callback.

File: assignments/callbacks.js [hunk 6]
```diff
@@ -24,91 +24,58 @@ const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum','yo-yo', 'Gum'];
...
+console.log(sumNums(5, 6));
 
+// sumNums adds two numbers (x, y) and passes the result to the callback.
 
-function contains(item, list, cb){
- cb(list.includes(item));
- }
-console.log(items.includes('yo-yo'));
+function multiplyNumbers(x, y, cb) {
+  return x * y;
```
Comment (liam-mei) [inline assignments/callbacks.js:53]: Same here.  You're return the product of x and y but not passing it to the callback.

**Review by liam-mei — APPROVED** [review]

**Comment by liam-mei** [comment]
- [x] Array Methods 5/5
- [ ] Callbacks 2/5
- [x] Closures 1/1

--------------
Stretch

- [x] callback
- [ ] closure

-------------

Score: 1

**Pull request merged**